budgetportal/tests/test_irm_preprocessor.py

```diff
@@ -110,3 +110,21 @@ def test_get_row_implementors_just_one(self):
         self.assertEqual(implementors[1], "")
         self.assertEqual(implementors[2], "C")
         self.assertEqual(implementors[3], "")
+
+
+class PreprocessEmptyRowTestCase(unittest.TestCase):
+    def test_empty_rows_removed(self):
+        headers = BASE_HEADERS + [REPEATED_IMPLEMENTOR_HEADER] * 20
+        dataset = Dataset(headers=headers)
+        num_nonempty_rows = dataset.height
+
+        num_of_empty_rows = 100
+        empty_row = [None] * len(headers)
+        for i in range(num_of_empty_rows):
+            dataset.append(empty_row)
+
+        total_rows = num_nonempty_rows + num_of_empty_rows
+        self.assertEqual(dataset.height, total_rows)
```

Comment:
please hardcode the expected values like we did before so that it's clearer when something is zero when it shouldn't be

Reply:
The expected values changed to hardcoded values.

https://github.com/vulekamali/datamanager/pull/213/commits/4b7fb0d949828d11deb02493dde613d4de3f13db